build_posix/configure.ac.in

```diff
@@ -29,18 +29,21 @@ define([AC_LIBTOOL_LANG_F77_CONFIG], [:])dnl
 # reason to believe "c++" can build compatible objects.
 #
 # Check whether the C++ compiler works by linking a trivial program.
-if test "$CC" = "cc"; then
-	AC_CACHE_CHECK([whether the C++ compiler works],
-		       [wt_cv_prog_cxx_works],
-		       [AC_LANG_PUSH([C++])
-			AC_LINK_IFELSE([AC_LANG_PROGRAM([], [])],
-				       [wt_cv_prog_cxx_works=yes],
-				       [wt_cv_prog_cxx_works=no])
-			AC_LANG_POP([C++])])
-else
-	AC_MSG_WARN([C++ compiler ignored unless compiler is named "cc"])
-	wt_cv_prog_cxx_works=no
-fi
+AM_CONDITIONAL([IS_CXX_OK], [test "$CC" = "cc"])
+# Compare gcc and g++ versions if required
+AM_COND_IF([IS_CXX_OK], [], AM_CONDITIONAL([IS_CXX_OK], [test $(expr `"$CC" --version | grep ^gcc | sed 's/^.* //g'` ) = $(expr `"$CXX" --version | grep ^g++ | sed 's/^.* //g'` )]))
```

Comment:
I don't think this is right yet.  On Linux:
```
$ CC=/usr/bin/cc
$ CXX=clang++
$ "$CC" --version
cc (GCC) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ "$CXX" --version
MongoDB clang version 7.0.1 (tags/RELEASE_701/final) (based on LLVM 7.0.1)
Target: x86_64-mongodb-linux
Thread model: posix
InstalledDir: /opt/mongodbtoolchain/v3/bin
$ test $(expr `"$CC" --version | grep ^gcc | sed 's/^.* //g'` ) = $(expr `"$CXX" --version | grep ^g++ | sed 's/^.* //g'` )
expr: missing operand
Try 'expr --help' for more information.
expr: missing operand
Try 'expr --help' for more information.
$ echo $?
0
```
 It looks like due to the expr errors, it ends up comparing "" to "".  The test succeeds (with 0), even though it shouldn't, clang c++ compiler won't work with gcc.   Developers may rename their compilers or use various pathnames.

Reply:
I wonder if this might work:
```
get_compiler_version() {
   # should work for gcc and clang variants.
   "$1" --version | egrep '(clang|GCC)' | sed -e 's/.*\((GCC)\)/\1/'
}
...
AM_COND_IF([IS_CXX_OK], [], AM_CONDITIONAL([IS_CXX_OK], [test "$(get_compiler_version "$CC")" != '' -a "$(get_compiler_version "$CC")" = "$(get_compiler_version "$CXX")"]))
```

Reply:
I updated the grep command, it should work with your configuration now